### PR TITLE
This enables us to use SSM to access instances

### PR DESCRIPTION
**Pull request by hammerdr**
* No additional charges for using SSM to access EC2 instances

https://aws.amazon.com/systems-manager/pricing/

### CODE REVIEW

File: aws/iam/instance_profile/main.tf [hunk 1]
```diff
@@ -24,6 +24,15 @@ resource "aws_iam_role" "mod" {
   path = "/"
 }
 
+data "aws_iam_policy" "AmazonSSMManagedInstanceCore" {
+  arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+resource "aws_iam_role_policy_attachment" "mod-ssm-role-policy" {
+  role       = aws_iam_role.mod.id
+  policy_arn = data.aws_iam_policy.AmazonSSMManagedInstanceCore.arn
```
Comment (phoolish) [inline aws/iam/instance_profile/main.tf:33]: why do you need the policy document if you are passing the ARN?

Comment (hammerdr) [reply aws/iam/instance_profile/main.tf:33]: the policy document above is just a reference to the aws one. We could just pass the arn direct, i think, but this puts a little more structure and validation around it. I have no strong feelings on this one, it was just what I researched this morning for using AWS managed policies in terraform :)

**Review by phoolish — APPROVED** [review]

**Comment by pturley** [comment]
@hammerdr just FYI, we didnt put these profiles on every box. In fact they are kind of a new thing for us and therefore arent on a lot of the fleet...just FYI

**Comment by hammerdr** [comment]
> @hammerdr just FYI, we didnt put these profiles on every box. In fact they are kind of a new thing for us and therefore arent on a lot of the fleet...just FYI

Yeah, this is just step one of a many, many step process to getting to a spot where we could actually deprecate SSH access. But, this seemed to be the way to codify a direction that we might want to pursue.

The other part of this is here: https://github.com/tablexi/txi-cli/commit/25bca19171add0648d9036d1286b6378e9cd7d5b

The general plan for a migration like this is something like:

1. Proof of Concept (this)
2. Decision to follow this plan
3. Survey of all places this needs to change, and changing them in a non-destructive way (e.g. add SSM next to SSH)
4. Announce planned deprecation broadly, write documentation for how to live in the new world
5. (wait)
6. Deprecate old way of doing things

**Comment by vandrijevik** [comment]
@hammerdr Should this PR be merged? It looks like it was applied to some Terraform code we have, and upgrading those uses of `terraform_modules` to a recent version shows a plan to undo the changes made by this PR…

**Comment by pturley** [comment]
@vandrijevik we talked about this and we are going to remove the usage of this so far and @hammerdr can add this in later once he has the time.

**Comment by vandrijevik** [comment]
@pturley Cool, thanks for the update!

**Comment by pturley** [comment]
Closing this until we can get an owner to take it up

**Pull request closed without merging**